src/pages/api/chooseValidity.ts

```diff
@@ -46,29 +73,27 @@ export default (req: NextApiRequestWithSession, res: NextApiResponse): void => {
             throw new Error('session is invalid.');
         }
 
-        if (req.body.validityInput === '0') {
-            setSession(req, res, 'The value of days your product is valid for cannot be 0.');
-            return;
-        }
+        const { validityInput, duration } = req.body;
+        const errorInfo: ErrorInfo[] = [];
 
-        if (!req.body.validityInput) {
-            setSession(req, res, 'The value of days your product is valid for cannot be empty.');
-            return;
-        }
+        errorInfo.concat(validityInputErrorCheck(errorInfo, validityInput));
 
-        if (isInvalidValidityNumber(req)) {
-            setSession(
-                req,
-                res,
-                'The value of days your product is valid for has to be a whole number between 1 and 366.',
-            );
-            return;
+        if (!duration || duration !== ('day' || 'week' || 'month' || 'year')) {
```

Comment:
For whatever reason, this condition never evaluates to false for 'week', 'month' or 'year'. Do we definitely need to check this here?

Reply:
I believe that is working as intended, it is not meant to evaluate to false for week month or year. Its because the user could edit the html and create their own stuff, so if we dont check it, it will end up in the Netex.

Reply:
When I select week, month or year on the page it doesn't work because of this check. I agree we should keep it in though if we want to stop jibberish getting in the netex